test/noInapplicableLifeCycleRule.spec.ts

```diff
@@ -0,0 +1,163 @@
+      `;
+      assertAnnotated({
+        ruleName, source,
+        message: 'The App class is not a Component or Directive yet implements these life cycle hooks: OnInit, OnDestroy.',
+      });
+    });
+    it(`should pass when class is a component`, () => {
+      const source1 = `@Component() class App implements OnInit, OnDestroy { }`;
+      const source2 = `@Component() @Injectable() class App implements OnInit, OnDestroy { }`;
+      assertSuccess(ruleName, source1);
+      assertSuccess(ruleName, source2);
+    });
+    it(`should fail when class is a service`, () => {
+      const source = `
+      @Injectable() class App implements OnInit, OnDestroy { }
+                          ~~~
+      `;
+      assertAnnotated({
+        ruleName, source,
+        message: 'The App class is not a Component or Directive yet implements a life cycle hook OnInit.',
+      });
+    });
+  });
+
+  describe(`inheritance`, () => {
+    it(`should pass a basic case`, () => {
+      const source = `
+                   class Base implements OnInit, OnDestroy { }
+      @Component() class Drvd extends Base { }
+      `;
+      assertSuccess(ruleName, source);
+    });
+    it(`should pass a more complex case`, () => {
+      const source = `
+                    class Base implements OnDestroy { }
+      @Component()  class Cmp1 extends Base implements OnInit, OnChanges { }
+      @Component()  class Cmp2 extends Base implements OnInit { }
+      @Injectable() class Srv1 extends Base { }  
+      `;
+      assertSuccess(ruleName, source);
+    });
+
+    it(`should fail a basic case`, () => {
+      // Note that this doesn't fail in `Base`, even though it would fail if it wasn't extended.
+      const source = `
+                    class Base implements OnInit, OnDestroy { }
+      @Injectable() class Drvd extends Base { }
+                          ~~~~
+      `;
+      const message = `The Drvd class is not a Component or Directive yet implements a life cycle hook OnInit (from Base).`;
+      assertAnnotated({ruleName, source, message});
+    });
+    it(`should fail a more complex case`, () => {
+      // Note that this doesn't fail in `VeryBase` or `Base`, even though it would fail if it wasn't extended.
+      const source = `
+                         class VeryBase implements OnInit, MyInterface, OnDestroy { }
+                         class Base     extends VeryBase implements AfterViewInit { }
+      @CustomDecorator() class A        extends Base { }
+                               ~
+      `;
+      const message = `The A class is not a Component or Directive yet implements these life cycle hooks: ` +
+        `OnInit (from Base -> VeryBase), OnDestroy (from Base -> VeryBase), AfterViewInit (from Base).`;
+      assertAnnotated({ruleName, source, message});
+    });
+  });
+
+  describe(`multiple inheritance paths`, () => {
+    // TODO: merge these two in a single test (blocked by assert annotated working with multiple failures)
```

Comment:
Another alternative is what tslint does [here](https://github.com/palantir/tslint/blob/master/test/rules/jsdoc-format/jsdoc.js.lint).